New 0.62.0 changelog (#166)
react-native-community/releases: CHANGELOG.md

- Summary: Changelog: [iOS][added] Make RCTTiming TurboModule-compatible ([2b5f4de7eb](https://github.com/facebook/react-native/commit/2b5f4de7eb4e5453e91d4956bcb0f58fe37b6562) by [@RSNara](https://github.com/RSNara))
- Make RCTNativeAnimatedMoudle TurboModule-compatible ([791931454e](https://github.com/facebook/react-native/commit/791931454e12f265d8aa12d85ba5e1ca3c2df539) by [@RSNara](https://github.com/RSNara))
- RCTRedBox ability to add extra buttons ([fc80a51457](https://github.com/facebook/react-native/commit/fc80a5145719ccecbefc25bfedbad6224b54efab) by [@mmmulani](https://github.com/mmmulani))
- Make RCTAsyncLocalStorage TurboModule-compatible ([2b62bd7695](https://github.com/facebook/react-native/commit/2b62bd7695e568ff97797f161ef9f75a063eece2) by [@RSNara](https://github.com/RSNara))
- Make RCTAlertManager TurboModule-compatible ([da6274ba76](https://github.com/facebook/react-native/commit/da6274ba76afaa271d6a74e4143479c93fa9bb14) by [@RSNara](https://github.com/RSNara))

> - Add definition for `queryCache` in ImageLoader ([ea1e8bbd34](https://github.com/facebook/react-native/commit/ea1e8bbd34b655a062ef13b6c2c76660ed43d31a) by [@lunaleaps](https://github.com/lunaleaps))

Author: TBH I'm aware of some people actively using the JSI directly, so I'd prefer to keep this in.

# Changelog

## [v0.62.0]

### Breaking

- Removing <TextInput>'s `onTextInput` event ([3f7e0a2c96](https://github.com/facebook/react-native/commit/3f7e0a2c9601fc186f25bfd794cd0008ac3983ab) by [@shergin](https://github.com/shergin))
- Remove TextInput's `inputView` prop ([1804e7cbea](https://github.com/facebook/react-native/commit/1804e7cbea707a35046118090966a54031edfae8) by [@TheSavior](https://github.com/TheSavior))
- Animated: Remove `defaultProps` Parameter ([a70987cee2](https://github.com/facebook/react-native/commit/a70987cee24bcd027b9c4a5aa85dfd6a1aab74b3) by [@yungsters](https://github.com/yungsters))

> - RN: Delete `TextInput.selectionState` Prop ([2becdfd404](https://github.com/facebook/react-native/commit/2becdfd4041f7f28138ba3a61c03e17c06dc2e50) by [@yungsters](https://github.com/yungsters))

Reviewer: How can i fit this change?

Reviewer 2: @youngjuning I’m not sure what the question is. If you are asking about the linked change, then I would suggest asking the question on that commit instead; this PR is purely about the CHANGELOG.

- ScrollView: Fix flow type for StickyHeaderComponentType ([5d2de9e54e](https://github.com/facebook/react-native/commit/5d2de9e54e03373627428ba9511d9cbc616424bd) by [@TheSavior](https://github.com/TheSavior))
- Change ScrollView event flow types to support spreading ([dd3bf0545a](https://github.com/facebook/react-native/commit/dd3bf0545abf48f5abc471049e2ce2073ade1cd9) by [@TheSavior](https://github.com/TheSavior))
- Remove duplicate flow type for TouchableProps ([fb5276c189](https://github.com/facebook/react-native/commit/fb5276c1897982f0be96f2e7074d839ae8fffdc8) by [@TheSavior](https://github.com/TheSavior))
- Lists: Fix invalid prop usage in RNTesterExampleList ([da057749f8](https://github.com/facebook/react-native/commit/da057749f8be59516b7dcb5cb3a7bdb67c3d990c) by [@TheSavior](https://github.com/TheSavior))
- Enable no-useless-escape lint rule ([90977b0e00](https://github.com/facebook/react-native/commit/90977b0e00acc6b3263502017c27094392e89478) by [@motiz88](https://github.com/motiz88))

> - Fix throttling in scroll view ([4159e20146](https://github.com/facebook

Reviewer 3: This seems strange, what happened here?

Reviewer 2: Thanks for identifying these! I’ll plug them into the regression test suite and get them sorted 👍

Reviewer 2: Fixed in https://github.com/react-native-community/releases/pull/169 and updated this PR.

> - Fix throttling in scroll view ([4159e20146](https://github.com/facebook

Reviewer 3: Here too. Same as the below error but also missing an author?

Reviewer 2: Fixed in https://github.com/react-native-community/releases/pull/169 and updated this PR.

> - Fix throttling in scroll view ([4159e20146](https://github.com/facebook

Reviewer 3: This one shows up as a syntax error in Sublime because of the use of brackets in the commit message implying the beginning of a link. Not sure if it is actually a problem or not but I wanted to flag.

Reviewer 2: I suspect Sublime is trying to be helpful about there being a space between `[2/N]` and the subsequent open parenthesis? It renders as we want it to, at least on GitHub.

# Changelog

## v0.62.0

> This major release includes Flipper support by default, improved dark mode support, moving Apple TV to [react-native-tvos](https://github.com/react-native-community/react-native-tvos), and more. See the [blog post](https://reactnative.dev/blog/2020/03/24/version-0.62) for all of the highlights.

Reviewer 4: ```suggestion
This major release includes Flipper support by default, improved dark mode support, moving Apple TV to [react-native-tvos](https://github.com/react-native-community/react-native-tvos), and more. See the [blog post](https://reactnative.dev/blog/2020/03/26/version-0.62) for all of the highlights.
```

- Remove `accessibilityStates` property ([7b35f427fd](https://github.com/facebook/react-native/commit/7b35f427fd66cb0f36921b992095fe5b3c14d8b9) by [@marcmulcahy](https://github.com/marcmulcahy))

#### Android specific

- Fix setting keyboardType from breaking autoCapitalize ([233fdfc014](https://github.com/facebook/react-native/commit/233fdfc014bb4b919c7624c90e5dac614479076f) by [@safaiyeh](https://github.com/safaiyeh))
- Limit size of video uploaded from camera roll in android (< 100 MB) ([d21f695edf](https://github.com/facebook/react-native/commit/d21f695edf367166a03af4c6e9376cd498b38665))
- Remove "Reload on JS change" from RN Android ([478df155e7](https://github.com/facebook/react-native/commit/478df155e70a4ce30219adcac6f0801c4e4d10ec) by [@cpojer](https://github.com/cpojer))

> #### iOS specific

Reviewer 4: this is empty, so remove?

# Changelog

## v0.62.0

This major release includes Flipper support by default, improved dark mode support, moving Apple TV to react-native-tvos, and more. See the [blog post](https://reactnative.dev/blog/2020/03/24/version-0.62) for all of the highlights.

This release comes in the midst of a global pandemic. We’re releasing this version today to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.

If you're upgrading, manual intervention may be required for your app. Please see the [upgrade-helper](https://react-native-community.github.io/upgrade-helper/) for a detailed breakdown of the changes required and see [this issue](https://github.com/react-native-community/releases/issues/179) for known issues.

Note two known issues with their corresponding workarounds:	

> - [Release version instacrash on Android](https://github.com/facebook/react-native/issues/28163)	

Reviewer 2: Btw, 👴 here, is ‘instacrash’ a word that helps people understand this sentence?

Author: I just copied the title of the issue, if there's a better way of phrasing it we could change it here and in the issue.

Maybe something like `Crash on Android when creating a release version`?

Reviewer 2: What about?

```suggestion
- [Crash on launch of Android release builds](https://github.com/facebook/react-native/issues/28163)	
```

Reviewer 2: Hmm, in retrospect, I thought this was about another issue–namely https://github.com/facebook/react-native/issues/28270.

I actually don’t think we should link to https://github.com/facebook/react-native/issues/28163, as it does not appear to be a problem affecting multiple people. All commenters seem to have different crash logs and the original poster seems to be using Vue (?).

Reviewer 2: Alright, removed this 👍

- Fix close button issue in KeyboardAvoidingView ([f1c6029e48](https://github.com/facebook/react-native/commit/f1c6029e4868084e5a10d81c15ee3cc5e301599a) by [@saxenanickk](https://github.com/saxenanickk))
- Fix activity recreation on theme change ([83a16b16c9](https://github.com/facebook/react-native/commit/83a16b16c9afa0fe0328ab818470d4fce098876b) by [@Esemesek](https://github.com/Esemesek))
- Fix ForwardingCookieHandler missing WebView exceptions. ([314eba98b2](https://github.com/facebook/react-native/commit/314eba98b2f2755cb26ed7a268d3fe83a7626efa) by [@siddhantsoni](https://github.com/siddhantsoni))
- Fix ReactInstanceManagerBuilder.build crashing if SoLoader has not been explicitly initialized ([60e00d9d96](https://github.com/facebook/react-native/commit/60e00d9d96d7b186c1d4c1542caddc1b74eeb3da) by [@petterh](https://github.com/petterh))
- Fix default accessibility hint not being read. ([f8dff0bcb3](https://github.com/facebook/react-native/commit/f8dff0bcb3147b7a1aa8ac7159952d848e198e29))
- Fix JS bundle loading progress bar ([7b9d6d19e2](https://github.com/facebook/react-native/commit/7b9d6d19e2c0854aa53587ef68ce715fb7803e2a) by [@makovkastar](https://github.com/makovkastar))

> - AFix ndroid Q related NaN error - don't try to do math with NaN values ([db5994980d](https://github.com/facebook/react-native/commit/db5994980df136c5cce6cd90348b4bf18180562f))

Reviewer 5: A small typo here.
```suggestion
- Fix Android Q related NaN error - don't try to do math with NaN values ([db5994980d](https://github.com/facebook/react-native/commit/db5994980df136c5cce6cd90348b4bf18180562f))
```

Reviewer 2: Thanks!

# Changelog

## v0.62.0

This feature release of React Native includes many changes for the platform: it has been a lot of work, so please refer to the [blog post](ADD) for overlook of the main changes - like Flipper support and the new LogBox.

> As always, we’ve done our best to test this release at length, especially with this being such a large release. However, note that due to challenges caused by COVID-19 to people everywhere, we [the maintainers] may not have the bandwidth we’d like to have for quick follow-up on bugs and/or a patch release. When you run into an issue, please do try to provide a fix or a detailed ticket, but be prepared to revert to the version you were previously on to unblock yourself.

Author: @alloy just a super quick question, in this phrase 
> [the maintainers]

is this supposed to be a link to something or it was more like " we = the maintainers"?

Reviewer 2: The latter 👍

- In Binding.cpp, retain reference to reactNativeConfig ([d7d848e824](https://github.com/facebook/react-native/commit/d7d848e8241ae430472ded5d428cacffe1e8bec7) by [@JoshuaGross](https://github.com/JoshuaGross))
- Remove hack in C++ that prevents any real TextInputs from being rendered ([219e1972d6](https://github.com/facebook/react-native/commit/219e1972d6f4ceeac4ffc41f8facc4205da78546) by [@JoshuaGross](https://github.com/JoshuaGross))
- Slightly more aggressive FabricUIManager teardown ([fa1c2be387](https://github.com/facebook/react-native/commit/fa1c2be387a95d90d8fcdaabb88b1c4c61e0097b) by [@JoshuaGross](https://github.com/JoshuaGross))
- Support for UpdatePaddingMountItem ([44bfc4be44](https://github.com/facebook/react-native/commit/44bfc4be447680096f01fd00a1344c2cd398a981) by [@JoshuaGross](https://github.com/JoshuaGross))
- Unbreak master ([a1ee1d7698](https://github.com/facebook/react-native/commit/a1ee1d7698081e7882d2fc5f08d848f93623f1c4) by [@JoshuaGross](https://github.com/JoshuaGross))

> - Work around crashing redbox ([43f042eb7d](https://github.com/facebook/react-native/commit/43f042eb7db0f

Reviewer 6: Should this change be under the `#### iOS Unknown` category?

Reviewer 7: The `Unknown` category isn’t actually part of the normal changelog - it’s items that haven’t been categorized yet. Ideally, all the `Unknown` items should all be gone when the changelog is finalized, so we’re not moving anything else into there

## v0.62.0

This feature release of React Native includes many changes for the platform: it has been a lot of work, so please refer to the [blog post](ADD) for overlook of the main changes - like Flipper support and the new LogBox.

> We also know that these past few weeks have been incredibly challenging for all of us, all over the world. The impact of COVID-19 is still reshaping our lives as we write these lines, and we hope that you will not feel pressured into upgrading to this latest. We went through more testing than ever to make sure that this new 62 will be a pleasant experience for everyone, but we are not perfect; so if you see a bug please remember that we are all in this together, and that it takes time for it to be addressed is because we all have higher priorities at this moment.

Reviewer 2: I’d be in favour of a little less appeasing and being a bit more matter of fact. For instance, something more to the point like:

```suggestion
As always, we’ve done our best to test this release at length, especially with this being such a large release. However, note that due to challenges caused by COVID-19 to people everywhere, we [the maintainers] may not have the bandwidth we’d like to have for quick follow-up on bugs and/or a patch release. When you run into an issue, please do try to provide a fix or a detailed ticket, but be prepared to revert to the version you were previously on to unblock yourself.
```

Author: cool - adding it.
